### PR TITLE
Doc for df getitem

### PR DESCRIPTION
**Pull request by Rubtsowa**

### CODE REVIEW

File: docs/source/api_ref/pandas.DataFrame.getitem.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+Limitations
+-----------
+Supported ``key`` can be one of the following:
+    - String literal, e.g. :obj:`df['A']`
+    - A slice, e.g. :obj:`df[2:5]`
+    - A tuple of string, e.g. :obj:`df[('A', 'B')]`
+    - An array of booleans, e.g. :obj:`df[True,False]`
+    - A series of booleans, e.g. :obj:`df(series([True,False]))`
+Column through getting attribute.
```
Comment (densmirn) [inline docs/source/api_ref/pandas.DataFrame.getitem.rst]:
```suggestion
Supported getting column through getting attribute.
```

File: docs/source/api_ref/pandas.DataFrame.getitem.rst [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+   :caption: Getting Pandas DataFrame column through getting attribute.
+   :name: ex_dataframe_getitem
+
+.. command-output:: python ./dataframe/df_getitem/df_getitem_key.py
+   :cwd: ../../../examples
+
+.. literalinclude:: ../../../examples/dataframe/df_getitem/df_getitem_column.py
+   :language: python
+   :lines: 27-
+   :caption: Getting Pandas DataFrame elements. Key is string.
```
Comment (densmirn) [inline docs/source/api_ref/pandas.DataFrame.getitem.rst]:
```suggestion
   :caption: Getting Pandas DataFrame column where key is a string.
```

File: docs/source/api_ref/pandas.DataFrame.getitem.rst [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    - String literal, e.g. :obj:`df['A']`
+    - A slice, e.g. :obj:`df[2:5]`
+    - A tuple of string, e.g. :obj:`df[('A', 'B')]`
+    - An array of booleans, e.g. :obj:`df[True,False]`
+    - A series of booleans, e.g. :obj:`df(series([True,False]))`
+Column through getting attribute.
+
+Examples
+--------
+.. literalinclude:: ../../../examples/dataframe/df_getitem/df_getitem_key.py
```
Comment (densmirn) [inline docs/source/api_ref/pandas.DataFrame.getitem.rst]:
```suggestion
.. literalinclude:: ../../../examples/dataframe/df_getitem/df_getitem_attr.py
```

File: docs/source/api_ref/pandas.DataFrame.getitem.rst [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+   :language: python
+   :lines: 27-
+   :caption: Getting Pandas DataFrame column through getting attribute.
+   :name: ex_dataframe_getitem
+
+.. command-output:: python ./dataframe/df_getitem/df_getitem_key.py
+   :cwd: ../../../examples
+
+.. literalinclude:: ../../../examples/dataframe/df_getitem/df_getitem_column.py
```
Comment (densmirn) [inline docs/source/api_ref/pandas.DataFrame.getitem.rst]:
```suggestion
.. literalinclude:: ../../../examples/dataframe/df_getitem/df_getitem.py
```

File: docs/source/api_ref/pandas.DataFrame.getitem.rst [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+   :caption: Getting Pandas DataFrame elements. Key is string.
+   :name: ex_dataframe_getitem
+
+.. command-output:: python ./dataframe/df_getitem/df_getitem_column.py
+   :cwd: ../../../examples
+
+.. literalinclude:: ../../../examples/dataframe/df_getitem/df_getitem_slice.py
+   :language: python
+   :lines: 27-
+   :caption: Getting Pandas DataFrame elements. Key is slice of integer.
```
Comment (densmirn) [inline docs/source/api_ref/pandas.DataFrame.getitem.rst]:
```suggestion
   :caption: Getting slice of Pandas DataFrame.
```

File: docs/source/api_ref/pandas.DataFrame.getitem.rst [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+Examples
+--------
+.. literalinclude:: ../../../examples/dataframe/df_getitem/df_getitem_key.py
+   :language: python
+   :lines: 27-
+   :caption: Getting Pandas DataFrame column through getting attribute.
+   :name: ex_dataframe_getitem
+
+.. command-output:: python ./dataframe/df_getitem/df_getitem_key.py
```
Comment (densmirn) [inline docs/source/api_ref/pandas.DataFrame.getitem.rst]:
```suggestion
.. command-output:: python ./dataframe/df_getitem/df_getitem_attr.py
```

File: docs/source/api_ref/pandas.DataFrame.getitem.rst [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+   :cwd: ../../../examples
+
+.. literalinclude:: ../../../examples/dataframe/df_getitem/df_getitem_column.py
+   :language: python
+   :lines: 27-
+   :caption: Getting Pandas DataFrame elements. Key is string.
+   :name: ex_dataframe_getitem
+
+.. command-output:: python ./dataframe/df_getitem/df_getitem_column.py
```
Comment (densmirn) [inline docs/source/api_ref/pandas.DataFrame.getitem.rst]:
```suggestion
.. command-output:: python ./dataframe/df_getitem/df_getitem.py
```

File: docs/source/api_ref/pandas.DataFrame.getitem.rst [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+   :caption: Getting Pandas DataFrame elements. Key is slice of integer.
+   :name: ex_dataframe_getitem
+
+.. command-output:: python ./dataframe/df_getitem/df_getitem_slice.py
+   :cwd: ../../../examples
+
+.. literalinclude:: ../../../examples/dataframe/df_getitem/df_getitem_tuple.py
+   :language: python
+   :lines: 27-
+   :caption: Getting Pandas DataFrame elements. Key is tuple of string.
```
Comment (densmirn) [inline docs/source/api_ref/pandas.DataFrame.getitem.rst]:
```suggestion
   :caption: Getting Pandas DataFrame elements where key is a tuple of strings.
```

File: docs/source/api_ref/pandas.DataFrame.getitem.rst [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+   :name: ex_dataframe_getitem
+
+.. command-output:: python ./dataframe/df_getitem/df_getitem_tuple.py
+   :cwd: ../../../examples
+
+.. literalinclude:: ../../../examples/dataframe/df_getitem/df_getitem_array.py
+   :language: python
+   :lines: 27-
+   :caption: Getting Pandas DataFrame elements. Key is array of boolean.
```
Comment (densmirn) [inline docs/source/api_ref/pandas.DataFrame.getitem.rst]:
```suggestion
   :caption: Getting Pandas DataFrame elements where key is an array of booleans.
```

File: docs/source/api_ref/pandas.DataFrame.getitem.rst [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+   :name: ex_dataframe_getitem
+
+.. command-output:: python ./dataframe/df_getitem/df_getitem_array.py
+   :cwd: ../../../examples
+
+.. literalinclude:: ../../../examples/dataframe/df_getitem/df_getitem_series.py
+   :language: python
+   :lines: 27-
+   :caption: Getting Pandas DataFrame elements. Key is series of boolean.
```
Comment (densmirn) [inline docs/source/api_ref/pandas.DataFrame.getitem.rst]:
```suggestion
   :caption: Getting Pandas DataFrame elements where key is series of booleans.
```

File: examples/dataframe/df_getitem/df_getitem_array.py [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+import numpy as np
+from numba import njit
+
+
+@njit
+def dataframe_getitem():
+    df = pd.DataFrame({'A': [0, 1, 2, 3, 4],
+                       'B': [1, 2, 3, 4, 5],
+                       'C': [2, 3, 4, 5, 6]})
+    arr = np.array([i % 2 for i in range(len(df))], dtype=np.bool_)
```
Comment (densmirn) [inline examples/dataframe/df_getitem/df_getitem_array.py]: It looks complicated. Let's prepare explicit array of booleans.

File: examples/dataframe/df_getitem/df_getitem_array.py [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+    result = df[arr]
+
+    return result
```
Comment (densmirn) [inline examples/dataframe/df_getitem/df_getitem_array.py]: These line could be combined to a single one. It's valid for other examples.

File: docs/source/api_ref/pandas.DataFrame.getitem.rst [hunk 13]
```diff
@@ -0,0 +1,74 @@
+.. _pandas.DataFrame.getitem:
+
+:orphan:
+
+pandas.DataFrame.getitem
+***************
+
+Evaluation obj[key]
```
Comment (densmirn) [inline docs/source/api_ref/pandas.DataFrame.getitem.rst]:
```suggestion
Evaluation of `obj[key]`
```

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 14]
```diff
@@ -1469,6 +1469,100 @@ def _df_getitem_bool_array_idx_impl(self, idx):
...
 def sdc_pandas_dataframe_getitem(self, idx):
+    """
+        Intel Scalable Dataframe Compiler User Guide
+        ********************************************
+        Pandas API: pandas.DataFrame.getitem
+
+        Evaluation of `obj[key]`
+
+
+
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]: Are you not able to decrease number of empty lines? If not, why?

File: docs/source/buildscripts/sdc_object_utils.py [hunk 15]
```diff
@@ -176,7 +176,10 @@ def get_obj(obj_name):
 
     split_name.pop(0)
     for name in split_name:
-        split_obj = getattr(split_obj, name)
+        if name in dir(split_obj):
+            split_obj = getattr(split_obj, name)
```
[the remark this answers is not in the document]
Comment (Rubtsowa) [reply docs/source/buildscripts/sdc_object_utils.py]: No, if-else need, otherwise the documentation will be uncorrectly.

**Pull request merged**